src/main/java/com/launchdarkly/client/FeatureFlag.java

```diff
@@ -70,36 +70,37 @@ EvalResult evaluate(LDUser user, FeatureStore featureStore) {
     return evaluate(user, featureStore, prereqEvents, visited);
   }
 
+  // Returning either a nil EvalResult or EvalResult.value indicates prereq failure/error.
   private EvalResult evaluate(LDUser user, FeatureStore featureStore, List<FeatureRequestEvent> events, Set<String> visited) {
```

Comment:
Changed logic a bit so we always send feature request events for all prereqs unless there is a data integrity problem: ie prereq cycle, or a prereq is not found in the feature store.